### PR TITLE
feat(funnels): Redesigned funnel viz

### LINKED ISSUE #8159: Would like to be able to see all of a funnel in the dashboard view with no horizontal scrollbar
## Is your feature request related to a problem?

I'd like to be able to see all the steps of a funnel in the dashboard view so my problem is that this isn't currently possible.

## Describe the solution you'd like

To be able to see a funnel in the dashboard view without a horizontal scroll bar.

## Describe alternatives you've considered

 I tried putting the funnel on vertical view, but this didn't seem to propagate to the dashboard view.

## Additional context

This is the current situation, I can only see two steps of the funnel.
image

### LINKED ISSUE #8990: Improve funnels layout for small screens
## Bug description

The funnels layout seems a bit cumbersome if you don't have a large monitor as the actual funnel visualization takes very little space (even on a 14-inch laptop). It's hard to actually interact with the funnel (I see at most one step) and particularly the table at the bottom.

Because the table at the bottom also contains the links to open the person modal (particularly for drop-offs), it's critical to be able to understand why my users aren't converting, and is not very intuitive that from here I can jump to watch recordings of why users dropped.

## Expected behavior

I would expect to see at least 3-4 steps without scrolling and have clear indications that I can deep dive into drop-offs to understand why users didn't convert.

## How to reproduce

1. Just create any funnel

## Environment

- [x] PostHog Cloud
- [x] self-hosted PostHog (ClickHouse-based), version/commit: _please provide_

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

Resolves #8159 and resolves #8990.

## Changes

Introduces new funnel viz that's much more effective at displaying results. [Design by Chris in Figma.](https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=8161%3A46841)
Feature-flagged under `lemon-funnel-viz`.

To-dos:
- [x] Add tooltip to bars
- [x] Hook up persons modal to bars
- [ ] Improve column sizing
- [x] Remove feature flag `lemon-funnel-viz` + old `FunnelStepTable` code
- [x] Improve color palette
- [x] Indicate scrollability
- [x] Fix sizing in `InsightCard`

_Screenshots in comment below._

### CODE REVIEW

**Comment by Twixes** [comment]
Updated screenshots:

## Page-level

### Before
before-main 

### After
after-main

## Card

### Before
before-card

### After
after-card

**Comment by Twixes** [comment]
This should be reviewable now. I put the new viz behind feature flag `lemon-funnel-viz` to avoid this PR getting even bigger. The remaining to-dos will be addressed in follow-up PRs.

**Review by mariusandra — APPROVED** [review]
Looks like code 👍

File: frontend/src/lib/components/LemonTable/LemonTable.tsx [hunk 1]
```diff
@@ -140,30 +137,7 @@ export function LemonTable<T extends Record<string, any>>({
...
-    })
-    useEffect(updateIsScrollable, [updateIsScrollable, width])
-    useEffect(() => {
-        const element = scrollRef.current
-        if (element) {
-            element.addEventListener('scroll', updateIsScrollable)
-            return () => element.removeEventListener('scroll', updateIsScrollable)
-        }
-    }, [updateIsScrollable])
+    const [scrollRef, scrollableClassNames] = useScrollable()
```
Comment (alexkim205) [inline frontend/src/lib/components/LemonTable/LemonTable.tsx:140]: lovely

**Review by alexkim205 — APPROVED** [review]
This is a huge improvement on the current steps table. Some considerations below that aren't blocking. Feel free to tackle in a separate PR as well.

1. We seem to stop assigning colors past a certain point (try $current_url breakdown).

Screen Shot 2022-04-14 at 9 49 26 AM

2. Long breakdown names hide rest of table.

Screen Shot 2022-04-14 at 9 52 37 AM

**Comment by clarkus** [comment]
image

What happens when the width of the step is larger than the width of the label? I feel like we need to base the sizing here on something consistent across steps. The chart feels broken because of the variable-width steps to me. It also would make it so steps with more verbose names could outcompete or displace adjacent steps.

**Comment by Twixes** [comment]
@mariusandra I take it you mean "Looks like good code" or "Looks like fine code"… because technically this is "code" too:
```javascript
function isEven(number) {
	if (number == 0) return true
	if (number == 1) return false
	if (number == 2) return true
	if (number == 3) return false
	if (number == 4) return true
	if (number == 5) return false
	if (number == 6) return true
	if (number == 7) return false
	if (number == 8) return true
	if (number == 9) return false
	if (number == 10) return true
}
```

@alexkim205 Thanks! Fixed both.

@clarkus Basically each step's column is as wide as its bars OR as wide as its label, whichever happens to be larger. I initially _wanted_ to do equal width columns, but that's very tricky to pull off with `table` elements while also accommodating to longer step labels (i.e. not truncating them) and keeping the first axis labels columns well-sized, so I went for the current scheme. As a side effect, this also ensures we never use more space for a single step than necessary (which is useful in insight cards particularly). I see how it can be jarring though. I can keep tinkering with `table` to see what can be done, but at least it doesn't seem to be a worse situation than what we've got now anyway.

**Comment by Twixes** [comment]
I'll merge this for now and keep sizing in mind for a follow-up. @clarkus If anything else is off, let me know.

File: frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelCorrelationTable.scss [hunk 2]
```diff
@@ -20,6 +20,7 @@
         border-top-left-radius: $radius;
         border-top-right-radius: $radius;
         padding-bottom: 5px;
+        overflow: auto;
```
Comment (neilkakkar) [inline frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelCorrelationTable.scss:23]: @Twixes do you remember why this was necessary? Seems to be the cause of the hidden popup

Comment (Twixes) [reply frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelCorrelationTable.scss:23]: Hmm, I think something was overlaid in a weird way without this, but honestly not sure at the moment. 🤔

**Pull request merged**